### PR TITLE
feat: dask expr `is_first|last_distinct` methods

### PR DESCRIPTION
**Pull request by FBruzzesi**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #637 

## Checklist

- [x] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

## If you have comments or can explain your changes, please do so below.

Dask does not implement Series/Expr `duplicated` as pandas. 
This implementation is very much inspired from the work on the pyarrow backend - I assume not very cheap though

### CODE REVIEW

File: narwhals/_dask/expr.py [hunk 1]
```diff
@@ -473,6 +473,50 @@ def clip(
...
+    def is_first_distinct(self: Self) -> Self:
+        def func(_input: Any) -> Any:
+            _name = _input.name
+            col_token = generate_unique_token(n_bytes=8, columns=[_name])
+            _input = (
+                _input.to_frame()
+                .assign(**{col_token: 1})
+                .assign(
+                    **{col_token: lambda t: t[col_token].cumsum(method="blelloch") - 1}
+                )
```
Comment (MarcoGorelli) [inline narwhals/_dask/expr.py:485]: is there a way to do this without passing a lambda to dask? been told it blocks some optimisations

**Review by MarcoGorelli — COMMENTED** [review]
looks good, just a comment (which might be applicable in other places)

File: narwhals/_dask/expr.py (hunk 1 above)
Comment (FBruzzesi) [reply narwhals/_dask/expr.py:485]: I will look into it, thanks for the heads up!

Comment (MarcoGorelli) [reply narwhals/_dask/expr.py:485]: thanks!

this is what I meant: https://github.com/narwhals-dev/narwhals/pull/765

It's fine if we have some lambdas on the Narwhals side which we evaluate before passing the result to Dask. But the input to Dask should ideally not be a lambda

Comment (FBruzzesi) [reply narwhals/_dask/expr.py]: Added and factored it out (3 times is enough to make a tiny function)

**Review by MarcoGorelli — APPROVED** [review]
legeeeend

**Pull request merged**